internal/scanner/walk.go

```diff
@@ -52,6 +53,8 @@ type Walker struct {
 	TempLifetime time.Duration
 	// If CurrentFiler is not nil, it is queried for the current file before rescanning.
 	CurrentFiler CurrentFiler
+	// If MtimeRepo is not nil, it is used to provide mtimes on systems that don't support setting arbirtary mtimes.
+	MtimeRepo *db.VirtualMtimeRepo
```

Comment:
Currently that is the case, but the code is written to handle it being nil.  In the initial version, this variable was an interface type so it made a bit more sense then.

Anyway, @calmh and I already had the discussion about short-circuiting the mechanism when it's not in use, but I think the conclusion is that a failed read is not so expensive to warrant that (yet).